weave-js/src/components/PagePanelComponents/Home/Browse3/pages/PlaygroundPage/PlaygroundChat/useChatFunctions.tsx

```diff
@@ -0,0 +1,154 @@
+import {SetStateAction} from 'react';
+
+import {Choice, Message} from '../../ChatView/types';
+import {OptionalTraceCallSchema, PlaygroundState} from '../types';
+import {DEFAULT_SYSTEM_MESSAGE} from '../usePlaygroundState';
+type TraceCallOutput = {
+  choices?: any[];
+};
+
+export const useChatFunctions = (
+  setPlaygroundStateField: (
```

Comment:
I found this a little hard to read, maybe pull out the type of the setPlaygroundStateField as an explicit type definition above?